dungeonGen.js

```diff
@@ -0,0 +1,284 @@
+			tempX = points[rmCent][0];
+			tempY = points[rmCent][1];
+			//randomly pick where the center of the room will be
+            switch (Math.round(Math.random() * 8)) {
+                case 0: { tempX += sizeOpts.tileSize; break; }
+                case 1: { tempX -= sizeOpts.tileSize; break; }
+           		case 2: { tempY += sizeOpts.tileSize; break; }
+                case 3: { tempY -= sizeOpts.tileSize; break; }
+                case 4: { tempX += sizeOpts.tileSize; tempY += sizeOpts.tileSize; break; }
+                case 5: { tempX -= sizeOpts.tileSize; tempY += sizeOpts.tileSize; break; }
+                case 6: { tempX += sizeOpts.tileSize; tempY -= sizeOpts.tileSize; break; }
+                case 7: { tempX -= sizeOpts.tileSize; tempY -= sizeOpts.tileSize; break; }
+                case 8: break;
+            }		
+
+            //build the room
+            for (var xind = tempX - sizeOpts.tileSize; xind <= tempX + sizeOpts.tileSize; xind += sizeOpts.tileSize) {
+                for (var yind = tempY - sizeOpts.tileSize; yind <= tempY + sizeOpts.tileSize; yind += sizeOpts.tileSize) {
+                	posit = [xind, yind];
+                	if(dungeonTiles.whereInStack(posit)==-1){
+	                    points.push(posit);
+	                    dungeonTiles.inStack.push(posit);
+                	}
+                }
+            }           	
+		}
+		dungeonTiles.color = 'rgb(255, 255, 255)';
+		tileOut.fillStyle = 'rgb(0,0,0)';
+		//draw the floor tiles
+		points.forEach(function(point) {
+			drawTile(point, sizeOpts.tileSize, dungeonTiles.color);
+		});
+		return points;
+	}
+
+	$('#creativeMode').click(function() {
+		$('#creMode').toggle(this.checked);
+	});
+
+	$('canvas').click(function(e) {
+		//records the coordinates of the canvas click
+		x = e.pageX - $('canvas').offset().left;
+		y = e.pageY - $('canvas').offset().top;
+		coords = {"xPos":x,"yPos":y};
+
+		if($('#creativeMode').is(':checked')) {
+			sizeChoice = $("input[name='size']:checked").val();
+			tSize = 10;
+			if(sizeChoice = 'small') 
+				tSize = 20;
+			else if(sizeChoice = 'medium')
+				tSize = 15;
+
+			x = Math.floor(x/tSize)*tSize;
+			y = Math.floor(y/tSize)*tSize;
+			posit = [x, y];
+
+			if($('input[name="edit"]:checked').val()=='add') {
+				//add tiles
+				dungeonTiles.color = 'rgb(255, 255, 255)';
+				if(dungeonTiles.whereInStack(posit)==-1) {
+					dungeonTiles.inStack.push(posit);
+				}
+				if(dungeonTiles.inStack.length*tSize*tSize >= 0.2*width*height)
+					$('#addButton').prop('disabled', true);
+				
+			}
+			else{
+				//remove tiles
+				dungeonTiles.color = 'rgb(0,0,0)';
+				if(dungeonTiles.whereInStack(posit)!=-1){ 
+					dungeonTiles.inStack.splice(dungeonTiles.whereInStack(posit), 1);
+				}
+				if(dungeonTiles.inStack.length*tSize*tSize < 0.2*width*height)
+					$('#addButton').prop('disabled', false);
+
+			}
+			drawTile(posit, tSize, dungeonTiles.color);			
+		}
+		return coords;
```

Comment:
I think the reason is that I need access to coords in other functions so I wanted its value to propagate to a one higher scope so other functions could use it.

Although I may be getting a bit confused since technically modifications in this function to coords should change the global value. I will try it without the return and see what happens